### PR TITLE
feature: add /block/:hash/raw api route

### PR DESCRIPTION
**Pull request by oleonardolima**
### Description
As decribed in #2180 this PR adds the /block/:hash/raw API route, it's now available when using mempool backend, with Bitcoin Core RPC.

In the production environment and using `esplora` as backend, it's still being ignored and must be redirected by the nginx configuration.

It also updates the doc message indicating that the type must differ by each backend used:
- esplora: block in binary representation
- bitcoin core rpc: block in hex

### CODE REVIEW

File: docker/backend/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:16.16.0-buster-slim AS builder
+FROM node:17.9.1-buster-slim AS builder
```
Comment (softsimon) [inline docker/backend/Dockerfile:1]: Is the change to the Dockerfile intended?

I don't see the relation to the PR so should be reverted.

**Review by softsimon — CHANGES_REQUESTED** [review]
Our goal for the Mempool API is to be identical to the Esplora API (when possible) so it doesn't matter which backend you run https://github.com/Blockstream/esplora/blob/master/API.md

So you need to change it so our bitcoin core based api returns the data in binary.

**Comment by oleonardolima** [comment]
> Our goal for the Mempool API is to be identical to the Esplora API (when possible) so it doesn't matter which backend you run https://github.com/Blockstream/esplora/blob/master/API.md
> 
> So you need to change it so our bitcoin core based api returns the data in binary.

Sure, I'll work on it

**Comment by oleonardolima** [comment]
> Our goal for the Mempool API is to be identical to the Esplora API (when possible) so it doesn't matter which backend you run https://github.com/Blockstream/esplora/blob/master/API.md
> 
> So you need to change it so our bitcoin core based api returns the data in binary.

@

> > Our goal for the Mempool API is to be identical to the Esplora API (when possible) so it doesn't matter which backend you run https://github.com/Blockstream/esplora/blob/master/API.md
> > So you need to change it so our bitcoin core based api returns the data in binary.
> 
> Sure, I'll work on it

@softsimon Sorry for the delay, I've updated it now 😁

**Comment by oleonardolima** [comment]
I've tested it locally with curl, postman, and with the following snippet:
``` rust
use std::ops::Deref;

use bitcoin::{consensus::deserialize, Block};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    env_logger::init();

    let hash = "1a72d2a7d9c7f0894b8e83344ec5d73e0c58588a7bee95f8552feb39423bca9e";
    let url = format!("http://localhost:8999/api/v1/block/{}/raw", hash);

    let res = reqwest::get(url).await?;
    log::debug!("{:#?}", res);

    let res_bytes = res.bytes().await?;
    log::debug!("{:#?}", res_bytes);
    let block: Block = deserialize(res_bytes.deref()).unwrap();
    // log::debug!("{:#?}", block);
    print!("{:#?}", block);

    Ok(())
}

```

File: backend/src/api/bitcoin/bitcoin.routes.ts [hunk 2]
```diff
@@ -470,6 +471,16 @@ class BitcoinRoutes {
     }
   }
 
+  private async getRawBlock(req: Request, res: Response) {
+    try {
+      const result = await bitcoinApi.$getRawBlock(req.params.hash);
+      res.setHeader('content-type', 'text/plain');
```
Comment (softsimon) [inline backend/src/api/bitcoin/bitcoin.routes.ts:477]:
```suggestion
      res.setHeader('content-type', 'application/octet-stream');
```

**Review by softsimon — APPROVED** [review]
tested ACK @ v2.5.0-dev [[a9248a5f](https://github.com/mempool/mempool/commit/a9248a5f)]

**Review by wiz — APPROVED** [review]
Tested ACK @ d66cc8a213f9e4f78197133ebc7db95f13617b4f

Output matches esplora exactly :+1:

**Pull request merged**